components/pedigree/resources/src/main/resources/pedigree/familyData.js

```diff
@@ -33,8 +33,12 @@ define([
                 this.familyMembersIndex[this.familyMembers[i].id] = i;
             }
 
-            this.warningMessage = familyJSON.hasOwnProperty("warning") ?
-                                  familyJSON.warning : "";
+            this.warningMessage = null;
+            if (familyJSON.hasOwnProperty("contains_sensitive_data") && familyJSON["contains_sensitive_data"]) {
```

Comment:
Personally I find the `hasOwnProperty` checks overzealous. I would have written lines 36-41 as:

```
this.warningMessage = familyJSON["contains_sensitive_data"] ? (familyJSON["sensitive_data_message"] || this.DEFAULT_WARNING_MESSAGE) : null;
```

Reply:
I think some hasOwnPropert can be removed, but I don't like to rely on pecularities of js such as `""` evaluating to false, it may not be obvious to every developer who looks at this. I may be wrong

Reply:
IMO, a dev who looks at this and doesn't know about truthy and falsy values in js should use the opportunity to learn something new.